framework/src/main/java/org/checkerframework/framework/flow/CFAbstractTransfer.java

```diff
@@ -1355,7 +1356,20 @@ protected V getWidenedValue(TypeMirror type, V annotatedValue) {
     public TransferResult<V, S> visitStringConversion(
             StringConversionNode n, TransferInput<V, S> p) {
         TransferResult<V, S> result = super.visitStringConversion(n, p);
-        result.setResultValue(p.getValueOfSubNode(n.getOperand()));
+        TypeMirror strType = n.getType();
+
+        Set<AnnotationMirror> operandAnnos = p.getValueOfSubNode(n.getOperand()).getAnnotations();
+        if (operandAnnos.isEmpty() && n.getOperand().getType().getKind() == TypeKind.TYPEVAR) {
```

Comment:
What about wildcards?
Can you add test cases for both type variables and wildcards?
This change has no impact for the existing type systems, so can you add a simple test type system that illustrates the problem? How would a type system writer e.g. ensure that the operand type is not a supertype of the String declaration bound?